main/realmain.cc

```diff
@@ -956,15 +956,25 @@ int realmain(int argc, char *argv[]) {
             vector<pair<string, int>> withNames;
             long sum = 0;
             for (auto e : untypedSources) {
-                withNames.emplace_back(core::SymbolRef::fromRaw(e.first).showFullName(*gs), e.second);
+                auto sym = core::SymbolRef::fromRaw(e.first);
+                if (!sym.exists() || !sym.loc(*gs).exists()) {
+                    continue;
+                }
+                auto path = sym.loc(*gs).file().data(*gs).path();
+                auto owner = sym.owner(*gs).show(*gs);
+                withNames.emplace_back(absl::StrCat(path, ",", owner, ",", sym.showFullName(*gs)), e.second);
                 sum += e.second;
             }
             fast_sort(withNames, [](const auto &lhs, const auto &rhs) -> bool { return lhs.second > rhs.second; });
             for (auto &p : withNames) {
-                logger->error("Typing `{}` would impact {}% callsites({} out of {}).", p.first, p.second * 100.0 / sum,
-                              p.second, sum);
+                logger->error("{},{},{}", p.first, p.second, sum);
```

Comment:
Slight anti-preference for CSV. We have method names with `,` in them:

https://sorbet.run/?arg=--print=symbol-table#%23%20typed%3A%20true%0Ait%20%22left%2Cright%22%20do%0Aend

I'd prefer doing JSON and generating it using the technique you can find in `tracing.cc`, using rapidjson.